src/test/java/com/uid2/shared/attest/AttestationTokenRetrieverTest.java

```diff
@@ -65,17 +64,42 @@ public void attest_succeed_attestationTokenSet(Vertx vertx, VertxTestContext tes
 
         when(clock.now()).thenReturn(Instant.parse("2023-08-01T00:00:00.111Z"));
 
-        attestationTokenRetriever.attest();
+        attestationTokenRetriever = getAttestationTokenRetriever(vertx);
+        testContext.awaitCompletion(500, TimeUnit.MILLISECONDS);
         Assertions.assertEquals("test_attestation_token", attestationTokenRetriever.getAttestationToken());
         Assertions.assertEquals("appName=appVersion;Component1=Value1;Component2=Value2", attestationTokenRetriever.getAppVersionHeader());
-        verify(this.responseWatcher, times(1)).handle(200);
+        verify(this.responseWatcher, times(1)).handle(Pair.of(200, expectedResponseBody));
         testContext.completeNow();
     }
 
     @Test
-    public void attest_currentTimeAfterTenMinsBeforeAttestationTokenExpiry_expiryCheckCallsAttest(Vertx vertx, VertxTestContext testContext) throws Exception {
+    public void attest_fixedIntervalUntilSuccess(Vertx vertx, VertxTestContext testContext) throws Exception {
+
+        when(attestationProvider.isReady()).thenReturn(true);
+        when(attestationProvider.getAttestationRequest(any())).thenReturn(new byte[1]);
+
+        HttpResponse<String> mockHttpResponse = mock(HttpResponse.class);
+        String expectedResponseBody = "{\"body\": {\"attestation_token\": \"test\",\"expiresAt\": \"2023-08-03T09:09:30.608597Z\",\"attestation_jwt_optout\": \"\",\"attestation_jwt_core\": \"\"},\"status\": \"success\"}";
+        when(mockHttpResponse.statusCode()).thenReturn(401, 401, 401, 200);
+        when(mockHttpResponse.body()).thenReturn("bad" ,"bad", "bad", expectedResponseBody);
+
+        when(mockHttpClient.send(any(HttpRequest.class), any(HttpResponse.BodyHandler.class))).thenReturn(mockHttpResponse);
+        when(mockAttestationTokenDecryptor.decrypt(any(), any())).thenReturn("test_attestation_token".getBytes(StandardCharsets.UTF_8));
+
+        when(clock.now()).thenReturn(Instant.parse("2023-08-01T00:00:00.111Z"));
+
         attestationTokenRetriever = getAttestationTokenRetriever(vertx);
```

Comment:
maybe we should reduce the retry interval for the test. This test will take 3 seconds every time

Reply:
reduced the interval